### PR TITLE
Multiple mapping task with same mapping url

### PR DESCRIPTION
**Pull request by Okanmercan99**

### CODE REVIEW

**Comment by Okanmercan99** [comment]
image
For this update, you should add a `name` field for each `mappingTask`. There should be no duplicate `mappingTask` names in a job definition, otherwise, `tofhir` will throw an error at the start.

File: tofhir-engine/src/main/scala/io/tofhir/engine/util/FhirMappingJobFormatter.scala [hunk 1]
```diff
@@ -59,7 +59,12 @@ object FhirMappingJobFormatter {
   def readMappingJobFromFile(filePath: String): FhirMappingJob = {
     val source = Source.fromFile(filePath, StandardCharsets.UTF_8.name())
     val fileContent = try replaceEnvironmentVariables(source.mkString) finally source.close()
-    org.json4s.jackson.JsonMethods.parse(fileContent).extract[FhirMappingJob]
+    val mappingJob = org.json4s.jackson.JsonMethods.parse(fileContent).extract[FhirMappingJob]
+    // check there are no duplicate name on mappingTasks of the job
+    if(!this.checkMappingTaskNamesUnique(mappingJob.mappings)){
+      throw new MappingException("Duplicate 'name' fields found in the MappingTasks within the MappingJob! Ensure each MappingTask has a unique name.")
```
Comment (dogukan10) [inline tofhir-engine/src/main/scala/io/tofhir/engine/util/FhirMappingJobFormatter.scala:65]: Add the duplicate name to the message.

File: tofhir-engine/src/main/scala/io/tofhir/engine/model/FhirMappingTask.scala [hunk 2]
```diff
@@ -142,6 +143,7 @@ object SourceFileFormats {
   final val AVRO = "avro"
   final val TXT_NDJSON = "txt-ndjson"
   final val TXT_CSV = "txt-csv"
+  final val TXT_TSV = "txt-tsv"
```
Comment (dogukan10) [inline tofhir-engine/src/main/scala/io/tofhir/engine/model/FhirMappingTask.scala:146]: Let's remove this.

File: tofhir-engine/src/main/scala/io/tofhir/engine/model/FhirMappingJob.scala [hunk 3]
```diff
@@ -41,11 +42,16 @@ case class FhirMappingJob(id: String = UUID.randomUUID().toString,
       throw new BadRequestException("Streaming jobs cannot be scheduled.")
     }
 
+    // Check names of the mappingTasks, if a duplicate name is found, throw an error
+    if(!FhirMappingJobFormatter.checkMappingTaskNamesUnique(mappings)){
+      throw new BadRequestException("Duplicate mappingTask name. Ensure each MappingTask has a unique name!")
```
Comment (dogukan10) [inline tofhir-engine/src/main/scala/io/tofhir/engine/model/FhirMappingJob.scala:47]: Add the duplicate name.

File: tofhir-engine/src/main/scala/io/tofhir/engine/mapping/MappingTaskExecutor.scala [hunk 4]
```diff
@@ -182,7 +188,7 @@ object MappingTaskExecutor {
           case (mappingExpr, resources, fhirInteraction) if fhirInteraction.exists(_.`type` == "patch") && resources.length > 1 =>
             Seq(FhirMappingResult(
               jobId = fhirMappingService.jobId,
-              mappingUrl = fhirMappingService.mappingUrl,
+              mappingTaskName = mappingTaskName,
```
Comment (dogukan10) [inline tofhir-engine/src/main/scala/io/tofhir/engine/mapping/MappingTaskExecutor.scala:191]: Previously, we were using `fhirMappingService.mappingUrl`, which is no longer in use. Can we replace `mappingUrl` with `mappingTaskName` in`fhirMappingService` and continue using it without modifying the method arguments, such as `executeMappingOnInput`?

File: tofhir-engine/src/main/scala/io/tofhir/engine/mapping/job/FhirMappingJobManager.scala [hunk 5]
```diff
@@ -422,13 +423,14 @@ class FhirMappingJobManager(
...
   def executeTask(jobId: String,
+                  mappingTaskName: String,
                   fhirMapping: FhirMapping,
                   df: DataFrame,
                   mainSourceSettings: MappingJobSourceSettings,
                   terminologyServiceSettings: Option[TerminologyServiceSettings] = None,
                   identityServiceSettings: Option[IdentityServiceSettings] = None,
                   executionId: Option[String] = None,
-                  projectId: Option[String] = None
+                  projectId: Option[String] = None,
```
Comment (dogukan10) [inline tofhir-engine/src/main/scala/io/tofhir/engine/mapping/job/FhirMappingJobManager.scala:433]: Comma is redundant.

**Review by dogukan10 — CHANGES_REQUESTED** [review]

**Review by dogukan10 — APPROVED** [review]

**Pull request merged**